src/formats/sgi.zig

```diff
@@ -0,0 +1,221 @@
+const Allocator = std.mem.Allocator;
+const buffered_stream_source = @import("../buffered_stream_source.zig");
+const color = @import("../color.zig");
+const FormatInterface = @import("../FormatInterface.zig");
+const ImageUnmanaged = @import("../ImageUnmanaged.zig");
+const ImageReadError = ImageUnmanaged.ReadError;
+const ImageError = ImageUnmanaged.Error;
+const utils = @import("../utils.zig");
+const std = @import("std");
+const PixelStorage = color.PixelStorage;
+const PixelFormat = @import("../pixel_format.zig").PixelFormat;
+const bigToNative = std.mem.bigToNative;
+
+pub const CompressionFlag = enum(u8) {
+    uncompressed = 0x0,
+    rle = 0x1,
+};
+
+pub const Dimensions = enum(u16) {
+    // one long scanline
+    scanline = 0x1,
+    // single channel bitmap, size = x_size * y_size
+    single_channel = 0x2,
+    // multi channel bitmap, size = z_size
+    multi_channel = 0x3,
+};
+
+pub const SgiPixelFormat = enum(u32) {
+    // normal: black&white = 1 chan, color = 3 chans, color + alpha = 4 chans
+    normal = 0x0,
+    // obsolete
+    dithered = 0x1,
+    // obsolete
+    need_cmap = 0x2,
+    // image is cmap-only
+    cmap_only = 0x3,
+};
+
+const Header = extern struct {
+    const size = 510;
+    const magic_number = [2]u8{ 0x1, 0xda };
+
+    compression: CompressionFlag align(1),
+    bits_per_channel: u8 align(1),
```

Comment:
whoups my bad this is more `bytes_per_channel`, clearly why `bpc` was a cofusing name, change it in the next PR